### PR TITLE
Add client program.

### PR DESCRIPTION
**Pull request by mstemm**
Add client program bin/juttle-engine-client, which inherits all the
capabilities of juttle-service-client and additionally has the
following commands:
 - browser --path <path>
 - push --path <path> [--topic <topic>]
 - watch --path <path> [--topic <topic>]

Thanks to the changes in juttle/juttle-service#22, this is easy as
defining some commands/cmdargs and implementing the commands in
functions. The functions were all copied from the latest
outrigger-client.

Also add the endpoint notifier back and add a handler for websocket
connections on /rendezvous/:topic to pass websocket connections to the
endpoint notifier. I also switched endpoint notifier to ES6 classes.

Add all necessary new packages to package.json. Although when I
tested, I used the master github version of juttle-servive, I didn't
check in that change. We'll need to update package.json to an
appropriate version of juttle-service.

@mnibecker I was thinking after this merges we should move the rendezvous endpoint below /api/v0. Now that it's separate from juttle-service, there's no confusion about what endpoints are official for juttle job management and what are for client/browser interactions.

@demmer @go-oleg

### CODE REVIEW

File: bin/juttle-engine-client [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+/* eslint no-console: 0 */
+var _ = require('underscore');
+var minimist = require('minimist');
+var JuttleServiceClient = require('juttle-service/lib/juttle-service-client');
+var Promise = require('bluebird');
+var WebSocket = require('ws');
+var path = require('path');
+var cli_errors = require('juttle/lib/cli/errors');
+var open = require('open');
+var JuttleBundler = require('juttle-service/lib/bundler');
```
Comment (go-oleg) [inline bin/juttle-engine-client]: A bit weary of reaching into `juttle-service` and grabbing these modules. My understanding is that not everything in `juttle-service-/lib` is _public_, but there is not distinction between what is and isn't. This means that when making changes to `juttle-service` like renaming or moving files, a developer will have no idea if its a breaking change and which semver version should be incremented.

Perhaps we can have one module within lib that exports everything that is public...or more everything that isn't public into some other directory?

File: bin/juttle-engine-client [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+var chokidar = require('chokidar');
+var fs = require('fs');
+var crypto = require('crypto');
+
+function push_file(filename, topic) {
+
+    return new Promise(function (resolve, reject) {
+
+        console.log('Bundling program ' + filename + ' and pushing to rendezvous topic ' + topic + '...');
```
Comment (go-oleg) [inline bin/juttle-engine-client]: could use some <3 from [template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/template_strings)

File: lib/endpoint-notifier.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+'use strict';
+// Simple echo server using websockets.
```
Comment (demmer) [inline lib/endpoint-notifier.js]: This looks like a stray comment

File: lib/endpoint-notifier.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+var _ = require('underscore');
+var logger = require('log4js').getLogger('endpoint-notifier');
```
Comment (demmer) [inline lib/endpoint-notifier.js]: This now needs to use `getLogger` from juttle-service.

**Comment by demmer** [comment]
Other than the above minor comments this looks good to me.

File: lib/endpoint-notifier.js (hunk 3 above)
Comment (mstemm) [inline lib/endpoint-notifier.js]: No, that's what it is, the endpoint notifier just echos messages from one websocket to all other websockets. I'll add a better comment saying that.

**Pull request merged**